src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -462,13 +462,6 @@ public static DetailAST parseWithComments(FileContents contents)
     }
 
     @Override
-    public void destroy() {
```

Comment:
> AbstractCheck#destroy is empty 

Did you read the comment there?
https://github.com/checkstyle/checkstyle/blob/ab548244eda102c5dc39cf0fc34e6803be192ba7/src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java#L135
https://github.com/checkstyle/checkstyle/blob/ab548244eda102c5dc39cf0fc34e6803be192ba7/src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheck.java#L65

I assuming these methods were added incase module had a file open and it needed to close it or free up other resources before ending.